packages/browser-wallet/src/popup/store/settings.ts

```diff
@@ -12,18 +13,17 @@ export const encryptedSeedPhraseAtom = atomWithChromeStorage<EncryptedData | und
 );
 export const themeAtom = atomWithChromeStorage<Theme>(ChromeStorageKey.Theme, Theme.Light);
 
-const storedJsonRpcUrlAtom = atomWithChromeStorage<string>(ChromeStorageKey.JsonRpcUrl, '', true);
-export const jsonRpcUrlAtomLoading = atom<AsyncWrapper<string>, string>(
-    (get) => get(storedJsonRpcUrlAtom),
-    (_, set, jsonRpcUrl) => {
-        set(storedJsonRpcUrlAtom, jsonRpcUrl);
-        popupMessageHandler.broadcast(EventType.ChainChanged, jsonRpcUrl);
-    }
+const storedNetworkConfigurationAtom = atomWithChromeStorage<NetworkConfiguration>(
+    ChromeStorageKey.NetworkConfiguration,
+    mainnet
 );
-export const jsonRpcUrlAtom = atom<string, string>(
```

Comment:
You could have made a selectAtom, which just selects the JsonRpcUrl from the networkConfiguration